bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/PageImpl.java

```diff
@@ -197,4 +200,20 @@ public boolean hasCloudconfigSupport() {
         }
         return hasCloudconfigSupport;
     }
+
+    @Override
+    public String getMainContentSelector() {
+        PageManager pageManager = request.getResourceResolver().adaptTo(PageManager.class);
+        if (pageManager == null) {
+            return Utils.getPolicyProperty(MAIN_CONTENT_SELECTOR_PROP, "", request);
+        }
+        com.day.cq.wcm.api.Page page = pageManager.getContainingPage(request.getResource());
+        if (page != null) {
+            String mainContentSelector = page.getProperties().get(MAIN_CONTENT_SELECTOR_PROP, "");
+            if (!mainContentSelector.isEmpty()) {
+                return mainContentSelector;
+            }
+        }
+        return Utils.getPolicyProperty(MAIN_CONTENT_SELECTOR_PROP, "", request);
```

Comment:
The content policy property can be read from the currentStyle (please have a look at PN_APP_RESOURCES_CLIENTLIB for example).